src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -299,47 +292,44 @@ bool TryGetFileMappingFromMetadataImporter(FileKey fileKey, [NotNullWhen(true)]
 
     /// <exception cref="IOException"/>
     /// <exception cref="BadImageFormatException" />
-    private static AssemblyMetadata CreateAssemblyMetadata(
-        FileKey fileKey,
-        Func<FileKey, (ModuleMetadata moduleMetadata, TemporaryStorageStreamHandle? storageHandle)> moduleMetadataFactory)
+    private static (AssemblyMetadata assemblyMetadata, IReadOnlyList<TemporaryStorageStreamHandle>? handles) CreateAssemblyMetadata(
```

Comment:
made static to make this easy to unit test.  this now returns both the metadta and handles to make this easy to unit tests.